extensions/_beacons/http_status.py

```diff
@@ -44,27 +47,22 @@ def validate(config):
         _config = {}
         list(map(_config.update, config))
 
-        if 'sites' not in _config:
-            return False, ('Configuration for %s beacon '
-                           'requires sites.', __virtualname__)
-        else:
-            if not isinstance(_config['sites'], dict):
-                return False, ('Sites for %s beacon '
-                               'must be a dict.', __virtualname__)
-            else:
-                for sites in _config['sites']:
-                    log.debug('_config %s', _config['sites'][sites])
-                    if 'url' not in _config['sites'][sites]:
-                        return False, ('Sites for %s beacon '
-                                       'requires url.', __virtualname__)
-                    if 'json_response' not in _config['sites'][sites]:
-                        return False, ('Sites for %s beacon '
-                                       'requires json_response.', __virtualname__)
-                    else:
-                        _json_response = _config['sites'][sites]['json_response']
-                        if not isinstance(_json_response, list):
-                            return False, ('json_response for %s beacon '
-                                           'must be a list.', __virtualname__)
+    try:
+        sites = _config.get('sites', {})
+    except AttributeError:
+        return False, ('Sites for %s beacon '
+                       'must be a dict.', __virtualname__)
+    if not sites:
+        return False, ('You neglected to define any sites')
+
+    for site, settings in sites.iteritems():
+        if required_site_attributes.isdisjoint(set(settings.keys())):
+            return False, ('Sites for {} beacon requires {}'.format(__virtualname__,
+                                                                    required_site_attributes))
+            if optional_site_attributes and optional_site_attributes.isdisjoint(set(settings.keys())):
```

Comment:
1) This is nested under the required attributes check
2) This will result in a failure if any of the optional attributes are absent from the settings which essentially makes them required...